qiskit_nature/second_q/operators/fermionic_op.py

```diff
@@ -14,158 +14,68 @@
-    ("-", "E"),
-    ("N", "E"),
-    ("E", "+"),
-    ("N", "-"),
-    ("E", "N"),
-}
-_MAPPING = {
-    ("I", "I"): "I",
-    ("I", "+"): "+",
-    ("I", "-"): "-",
-    ("I", "N"): "N",
-    ("I", "E"): "E",
-    ("+", "I"): "+",
-    ("+", "-"): "N",
-    ("+", "E"): "+",
-    ("-", "I"): "-",
-    ("-", "+"): "E",
-    ("-", "N"): "-",
-    ("N", "I"): "N",
-    ("N", "+"): "+",
-    ("N", "N"): "N",
-    ("E", "I"): "E",
-    ("E", "-"): "-",
-    ("E", "E"): "E",
-}
-
-
-class FermionicOp(SecondQuantizedOp):
-    r"""
-    N-mode Fermionic operator.
-
-    **Label**
-
-    Allowed characters for the label are `I`, `-`, `+`, `N`, and, `E`.
-
-    .. list-table::
-        :header-rows: 1
-
-        * - Label
-          - Mathematical Representation
-          - Meaning
-        * - `I`
-          - :math:`I`
-          - Identity operator
-        * - `-`
-          - :math:`c`
-          - Annihilation operator
-        * - `+`
-          - :math:`c^\dagger`
-          - Creation operator
-        * - `N`
-          - :math:`n = c^\dagger c`
-          - Number operator
-        * - `E`
-          - :math:`I - n = c c^\dagger`
-          - Hole number
-
-    There are two types of label modes for this class.
-    The label mode is automatically detected by the presence of underscore `_`.
-
-    1. Dense Label
-
-    Dense labels are strings with allowed characters above.
-    This is similar to Qiskit's string-based representation of qubit operators.
-    For example,
-
-    .. code-block:: python
-
-        "+"
-        "II++N-IE"
-
-    are possible labels.
-
-    2. Sparse Label
-
-    When the parameter `register_length` is passed to :meth:`~FermionicOp.__init__`,
-    label is assumed to be a sparse label.
-    A sparse label is a string consisting of a space-separated list of words.
-    Each word must look like :code:`[+-INE]_<index>`, where the :code:`<index>`
-    is a non-negative integer representing the index of the fermionic mode.
-    For example,
-
-    .. code-block:: python
+from .sparse_label_op import SparseLabelOp
 
-        "+_0"
-        "-_2"
-        "+_0 -_1 +_4 +_10"
 
-    are possible labels.
+class FermionicOp(SparseLabelOp):
+    r"""N-mode Fermionic operator.
 
-    As a programmatic approach, you can input a list of tuples as the labels,
-    too. Thus, a FermionicOp can be initialized using a list of tuples of the
-    form, `(action, index)`. `action` is a string where `-` denotes an
-    annihilation and `+` a creation operation. The `index` is also an integer
-    denoting the fermionic mode.
-    The following labels are equivalent to the example given for sparse labels
-    above:
-
-    .. code-block:: python
-
-        [("+", 0)]
-        [("-", 2)]
-        [("+", 0), ("-", 1), ("+", 4), ("+", 10)]
+    A `FermionicOp` represents a weighted sum of fermionic creation/annihilation operator terms.
+    These terms are encoded as strings consisting of a space-separated list of words. Each word must
```

Comment:
> These terms are encoded as strings consisting of a space-separated list of words

`words` here seems a bit odd - I had wondered about `expressions`. And expression would then be an operation and index. Or you could have these as operations where an operation comprises an operator creation/annihilation and an index. Though maybe we are trying to avoid too much overloaded use of  operator

Reply:
Thanks, that's a good suggestion! I think the old docstring also used `word` which is where I got that from :smile: